lib/req_bigquery.ex

```diff
@@ -83,29 +91,39 @@ defmodule ReqBigQuery do
       ...>  ORDER BY views DESC
       ...> \"""
       iex> req = Req.new() |> ReqBigQuery.attach(goth: MyGoth, project_id: project_id)
-      iex> Req.post!(req, bigquery: {query, ["Linkin_Park"]}).body
+      iex> res = Req.post!(req, bigquery: {query, ["Linkin_Park"]}).body
       %ReqBigQuery.Result{
         columns: ["year", "views"],
         job_id: "job_GXiJvALNsTAoAOJ39Eg3Mw94XMUQ",
         num_rows: 7,
-        rows: [[2017, 2895889], [2016, 1173359], [2018, 1133770], [2020, 906538], [2015, 860899], [2019, 790747], [2021, 481600]]
+        rows: %Stream{}
       }
+      iex> res.rows |> Enum.to_list()
+      [[2017, 2895889], [2016, 1173359], [2018, 1133770], [2020, 906538], [2015, 860899], [2019, 790747], [2021, 481600]]
 
   """
   @spec attach(Request.t(), keyword()) :: Request.t()
   def attach(%Request{} = request, options \\ []) do
+    checked_options =
+      options
+      |> Keyword.put_new(:base_url, @base_url)
+      |> Keyword.put_new(:max_results, @max_results)
+
     request
     |> Request.prepend_request_steps(bigquery_run: &run/1)
     |> Request.register_options(@allowed_options)
-    |> Request.merge_options(options)
+    |> Request.merge_options(checked_options)
   end
 
   defp run(%Request{options: options} = request) do
     if query = options[:bigquery] do
-      base_url = options[:base_url] || @base_url
+      base_url = options[:base_url]
       token = Goth.fetch!(options.goth).token
       uri = URI.parse("#{base_url}/projects/#{options.project_id}/queries")
-      json = build_request_body(query, options[:default_dataset_id])
+
+      json =
+        build_request_body(query, options[:default_dataset_id])
+        |> Map.put(:maxResults, options.max_results)
```

Comment:
```suggestion
      json =
        query
        |> build_request_body(options[:default_dataset_id])
        |> Map.put(:maxResults, options.max_results)
```
(we usually use pipe if there are more than a single function call :))